iothub_client/inc/iothub_client_properties.h

```diff
@@ -0,0 +1,298 @@
+    int ackVersion;
+    /** @brief Optional friendly description of the operation.  May be NULL. */
+    const char* description;
+} IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE;
+
+/** @brief Enumeration that indicates whether a given property from the service was originally reported from the device
+           (and hence the device application sees what the last reported value IoT Hub has)
+           or whether this is a writable property that the service is requesting configuration for.
+*/
+typedef enum IOTHUB_PROPERTY_TYPE_TAG 
+{
+    /** @brief Property was initially reported from the device. */
+    IOTHUB_CLIENT_PROPERTY_TYPE_REPORTED_FROM_DEVICE,
+    /** @brief Property is writable.  It is configured by service remotely. */
+    IOTHUB_CLIENT_PROPERTY_TYPE_WRITABLE
+} IOTHUB_CLIENT_PROPERTY_TYPE;
+
+/* @brief Enumeration that indicates whether the JSON value of a deserialized property 
+*         is a null-terminated string or binary.
+*/
+typedef enum IOTHUB_CLIENT_PROPERTY_VALUE_TYPE_TAG
+{
+    /* @brief Deserialized JSON value is a string. */
+    IOTHUB_CLIENT_PROPERTY_VALUE_STRING,
+    /* @brief Deserialized JSON value is a binary. */
+    IOTHUB_CLIENT_PROPERTY_VALUE_BINARY
+} IOTHUB_CLIENT_PROPERTY_VALUE_TYPE;
+
+/** @brief Current version of @p IOTHUB_CLIENT_DESERIALIZED_PROPERTY structure.  */
+#define IOTHUB_CLIENT_DESERIALIZED_PROPERTY_VERSION_1 1
+
+/** @brief    This struct represents a property from IoT Hub.  
+              It is generated by IoTHubClient_Deserialize_Properties while deserializing a property payload. */
+typedef struct IOTHUB_CLIENT_DESERIALIZED_PROPERTY_TAG {
+    /** @brief   Version of the structure.  Currently must be IOTHUB_CLIENT_DESERIALIZED_PROPERTY_VERSION_1. */
+    int structVersion;
+    /** @brief   Whether this is a property the device reported (and we're seeing what IoT Hub last received from 
+         the device) or whether it's a writable property request from the device. */
+    IOTHUB_CLIENT_PROPERTY_TYPE propertyType;
+    /** @brief Name of the component.  Optional; may be NULL for the root component. */
+    const char* componentName;
+    /** @brief Name of the property. */
+    const char* name;
+    /** @brief Whether the value is passed as a string or non-null terminated length. */
+    /** @description  Applications can safely assume that this will always be IOTHUB_CLIENT_PROPERTY_VALUE_STRING.  
+    *                 They do not need to write code to handle IOTHUB_CLIENT_PROPERTY_VALUE_BINARY.
+    *                 IOTHUB_CLIENT_PROPERTY_VALUE_BINARY may be supported by the SDK in the future, but 
+    *                 IOTHUB_CLIENT_PROPERTY_VALUE_STRING will *always* by the default.
+    */
+    IOTHUB_CLIENT_PROPERTY_VALUE_TYPE valueType;
+    /** @brief Value of the property. Currently this is only set to str. */
+    union {
+        const char* str;
+        unsigned const char* bin;
```

Comment:
If this is the byte case should this be
```suggestion
        const uint8_t* bin;
```